common/log.c

```diff
@@ -408,16 +408,35 @@ internal_log_config_dump(struct log_config *config)
     struct log_logger_level* logger;
 
     g_printf("logging configuration:\r\n");
-    internal_log_lvl2str(config->log_level, str_level);
-    g_printf("\tLogFile:       %s\r\n", config->log_file);
-    g_printf("\tLogLevel:      %s\r\n", str_level);
+    if (config->log_file)
+    {
+        internal_log_lvl2str(config->log_level, str_level);
+        g_printf("\tLogFile:       %s\r\n", config->log_file);
+        g_printf("\tLogLevel:      %s\r\n", str_level);
+    }
+    else
+    {
+        g_printf("\tLogFile:       %s\r\n", "<disabled>");
+    }
     
-    internal_log_lvl2str(config->console_level, str_level);
-    g_printf("\tEnableConsole: %s\r\n", (config->enable_console ? "true" : "false"));
+    if (config->enable_console)
+    {
+        internal_log_lvl2str(config->console_level, str_level);
+    }
+    else
+    {
+        g_strcpy(str_level, "<disabled>");
+    }
     g_printf("\tConsoleLevel:  %s\r\n", str_level);
-    
-    internal_log_lvl2str(config->syslog_level, str_level);
-    g_printf("\tEnableSyslog:  %s\r\n", (config->enable_syslog ? "true" : "false"));
+
+    if (config->enable_syslog)
+    {
+        internal_log_lvl2str(config->syslog_level, str_level);
+    }
+    else
+    {
+        g_strcpy(str_level, "<disabled>");
+    }
     g_printf("\tSyslogLevel:   %s\r\n", str_level);
     
     g_printf("per logger configuration:\r\n");
```

Comment:
Tangential to your work:

Reading this diff, I'm realizing that the per logger config may be confusing to xrdp admins since it is only relevant for XRDP_DEBUG builds. If you have time can you clean it up?

Reply:
Never mind about my comment here. Since this is tangential to your work, and I am also updating the logging code, I went ahead and implement removing the per logger config dump for non-XRDP_DEBUG builds in #1708